akka-persistence-tck/src/main/scala/akka/persistence/CapabilityFlags.scala

```diff
@@ -52,3 +52,9 @@ trait SnapshotStoreCapabilityFlags extends CapabilityFlags {
   // no flags currently
 }
 //#snapshot-store-flags
+
+trait CurrentEventsByTagCapabilityFlags extends CapabilityFlags {
```

Comment:
@Dennis. Well, I suppose this can be handled on a best effort basis.
Developers would have to read the previous journal and write to the new
journal. Depending on how the events are tagged or not, the new ordering
would be more or less successful. However, since they have already read the
old events somehow, the order they are in with the new journal is possibly
less important. In fact now that I think about it more, if they read the
old journal in the same way that they did previously, it could possibly
preserve order close to its original consumption.
On Jun 22, 2016 6:24 PM, "Dennis Vriend" <notifications@github.com> wrote:

In
akka-persistence-tck/src/main/scala/akka/persistence/CapabilityFlags.scala
<https://github.com/akka/akka/pull/20807#discussion_r68086177>:

> @@ -52,3 +52,9 @@ trait SnapshotStoreCapabilityFlags extends CapabilityFlags {
>    // no flags currently
>  }
>  //#snapshot-store-flags
> +
> +trait CurrentEventsByTagCapabilityFlags extends CapabilityFlags {

@VaughnVernon <https://github.com/VaughnVernon> The inmemory plugin also
has this 'problem'. For that plugin the solution is quite simple, just mark
each event with a unique incrementing ID for example the
java.util.concurrent.atomic.AtomicLong. For the RDBMS the database itself
is ultimately responsible for this. I'm just wondering how the migration
script would look like for current users :)

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/akka/akka/pull/20807/files/1e7268bb91b8975509db868825ac4c1f78b64c41#r68086177>,
or mute the thread
<https://github.com/notifications/unsubscribe/ACg87LT6yYNJ-iiGPZdt0mCi8cqEZDuCks5qOWHKgaJpZM4I5gQ6>
.